yt_dlp/extractor/svt.py

```diff
@@ -429,24 +469,27 @@ def suitable(cls, url):
     def _real_extract(self, url):
         path, display_id = self._match_valid_url(url).groups()
 
-        article = self._download_json(
-            'https://api.svt.se/nss-api/page/' + path, display_id,
-            query={'q': 'articles'})['articles']['content'][0]
-
+        webpage = self._download_webpage(url, display_id)
+        title = self._og_search_title(webpage)
         entries = []
 
-        def _process_content(content):
-            if content.get('_type') in ('VIDEOCLIP', 'VIDEOEPISODE'):
-                video_id = compat_str(content['image']['svtId'])
-                entries.append(self.url_result(
-                    'svt:' + video_id, SVTPlayIE.ie_key(), video_id))
+        urqlJson = self._search_json(r'window\.svt\.nyh\.urqlState[^{]+', webpage, 'video_id', display_id, default=None)
+
+        data = self._parse_json(traverse_obj(urqlJson, (..., 'data'), default=None, get_all=False), display_id)
+        video_ids = []
+        video_ids.append(traverse_obj(data, ('page', 'topMedia', 'svtId'), default=''))
+        video_ids.extend(traverse_obj(data, ('page', 'body', ..., 'video', 'svtId'), get_all=True, default=[]))
 
-        for media in article.get('media', []):
-            _process_content(media)
+        for video_id in video_ids:
+            if not video_id or len(video_id) == 0:
+                continue
 
-        for obj in article.get('structuredBody', []):
-            _process_content(obj.get('content') or {})
+            video_info = self._download_json('https://api.svt.se/video/' + video_id, video_id)
 
-        return self.playlist_result(
-            entries, str_or_none(article.get('id')),
-            strip_or_none(article.get('title')))
+            info_dict = self._extract_video(video_info, video_id)
+            info_dict.update({
+                'title': title
+            })
+            entries.append(info_dict)
+
+        return self.playlist_result(entries, display_id, title)
```

Comment:
for the playlist entries we can use a generator function with a single traversal, and we can use a set to eliminate duplicate IDs

```py
        def entries():
            for video_id in set(traverse_obj(data, (
                'page', (('topMedia', 'svtId'), ('body', ..., 'video', 'svtId')), {str}
            ))):
                info = self._extract_video(
                    self._download_json(f'https://api.svt.se/video/{video_id}', video_id), video_id)
                info.setdefault('title', title)
                yield info

        return self.playlist_result(entries(), display_id, title)
```

Reply:
`info.setdefault` didn't work, because the title key was already set as `None`